keyvi/src/cpp/dictionary/fsa/generator.h

```diff
@@ -51,12 +51,11 @@ namespace fsa {
  * @returns length of the longest common prefix of given strings
  */
 
-inline size_t get_common_prefix_length(const char* first, const char* second) {
+inline size_t get_common_prefix_length(const std::string& first, const std::string& second) {
 
   size_t common_prefix_length = 0;
 
-  while (first[common_prefix_length] == second[common_prefix_length]
-      && first[common_prefix_length] != 0) {
+  while (first[common_prefix_length] == second[common_prefix_length] && common_prefix_length < first.size()) {
```

Comment:
Is it possible that `first.size() > second.size()` ?
At least we can have an `assert` here.